File: drive-front/src/app/components/file/File.ts
export class FileClass {
  id: string;
  originalName: string;
  name: string
  size: string;
  createdDate: string;
  type: string;
  fileBytes: string[];
  isModalOpen: boolean;
  constructor(id: string, originalName: string, size: string, type: string, fileBytes: string[], createdDate: string, isModalOpen: boolean, name: string) {
    this.name = name
    this.isModalOpen = isModalOpen;
    this.createdDate = createdDate;
    this.id = id;
    this.originalName = originalName;
    this.size = size;
    this.type = type;
    this.fileBytes = fileBytes;
  }
}

File: drive-back/src/main/java/com/gabriel/drive_back/domain/File.java
package com.gabriel.drive_back.domain;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table
@Data
@NoArgsConstructor
@AllArgsConstructor
public class File {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String originalName;
    private String name;
    private Long size;
    private String createdDate;
    private String type;
    @Lob
    private byte[] fileBytes;

    public File(String originalName, Long size, String type, byte[] fileBytes, String createdDate, String name) {
        this.name = name;
        this.createdDate = createdDate;
        this.originalName = originalName;
        this.size = size;
        this.type = type;
        this.fileBytes = fileBytes;
    }
}
